core/src/components/datetime/utils/warn.ts

```diff
@@ -0,0 +1,52 @@
+import { printIonWarning } from '@utils/logging';
+
+import type { DatetimePresentation, FormatOptions } from '../datetime-interface';
+
+/**
+ * If a time zone is provided in the format options, the rendered text could
+ * differ from what was selected in the Datetime, which could cause
+ * confusion.
+ */
+export const warnIfTimeZoneProvided = (formatOptions?: FormatOptions) => {
+  if (
+    formatOptions?.date?.timeZone ||
+    formatOptions?.date?.timeZoneName ||
+    formatOptions?.time?.timeZone ||
+    formatOptions?.time?.timeZoneName
+  ) {
+    printIonWarning('Datetime: "timeZone" and "timeZoneName" are not supported in "formatOptions".');
```

Comment:
In most places that we use `printIonWarning`, we additionally log the element reference that is associated with the warning. This helps developers know which element is throwing the warning, when they have multiple on the view. 

I think it'd be worth passing the element reference into these functions (all in this file) and log the element reference as the second argument:

```ts
printIonWarning("Your message", el);
```

Reply:
updated in 85a7d73763

also added the HTML templates in 016b60a7c2